src/bitmap.c

```diff
@@ -2058,30 +2059,22 @@ GdipBitmapSetPixel (GpBitmap *bitmap, INT x, INT y, ARGB color)
 		pixel_format = data->pixel_format;
 	}
 
+	BYTE b, g, r, a;
+	get_pixel_bgra (color, b, g, r, a);
 	switch (pixel_format) {
 	case PixelFormat24bppRGB:
+		gdip_setpixel_24bppRGB (v, x, a, r, g, b);
+		break;
 	case PixelFormat32bppRGB:
-		color |= 0xFF000000; /* force the alpha for Cairo */
```

Comment:
removed this to get behaviour matching GDI+. If this breaks cairo, happy to revert.